src/main/java/com/mojang/datafixers/functions/PointFreeRule.java

```diff
@@ -590,6 +595,32 @@ public int hashCode() {
         }
     }
 
+    record Choice(PointFreeRule[] rules) implements PointFreeRule {
+        @Override
+        public <A> Optional<? extends PointFree<A>> rewrite(final Type<A> type, final PointFree<A> expr) {
+            for (final PointFreeRule rule : rules) {
+                final Optional<? extends PointFree<A>> view = rule.rewrite(type, expr);
+                if (view.isPresent()) {
+                    return view;
+                }
+            }
+            return Optional.empty();
+        }
+
+        @Override
+        public boolean equals(final Object obj) {
```

Comment:
This is a record, so we probably do not need explicit equals/hashCode anymore :)

Reply:
This one is unfortunately still required for the `Arrays.equals`/`Arrays.hashCode` :)

Reply:
Ah, I see, default `.equals` does not do what we want for arrays; one more reason to prefer `List<T>` over `T[]`, but not that important here :)